packages/assets-controllers/src/TokensController.ts

```diff
@@ -92,37 +94,63 @@ type SuggestedAssetMeta = {
  * @property allIgnoredTokens - Object containing hidden/ignored tokens by network and account
  * @property allDetectedTokens - Object containing tokens detected with non-zero balances
  */
-// This interface was created before this ESLint rule was added.
-// Convert to a `type` in a future major version.
-// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
-export interface TokensState extends BaseState {
-  tokens: Token[];
-  ignoredTokens: string[];
-  detectedTokens: Token[];
-  allTokens: { [chainId: Hex]: { [key: string]: Token[] } };
-  allIgnoredTokens: { [chainId: Hex]: { [key: string]: string[] } };
-  allDetectedTokens: { [chainId: Hex]: { [key: string]: Token[] } };
-}
+export type TokensState = BaseState &
+  Record<string, unknown> & {
```

Comment:
This was because `ControllerState` is constrained as `Record<string, unknown>` in base-controller, but `BaseState` is an interface, which TIL doesn't extend `Record<string, unknown>` unless the index signature is explicitly specified as string. 

Since we'll be removing `BaseState` for the v2 upgrade, I fixed this by moving the `BaseState` augmentation to the controller class generic argument: 25fe76d8c2cd8ffaca3a05fc22f301ede73f15c9.

That's a good catch that `string` should be a constraint for state keys, not an allow-all setting. I'll keep that in mind.

Reply:
Oh sorry — I mistakenly thought we were already using BaseControllerV2 for this controller. This makes sense.